src/components/RedirectionNotificationForUnsupportedBrowsers.js

```diff
@@ -0,0 +1,169 @@
+import * as s from "./RedirectionNotificationModal.sc";
+import CloseRoundedIcon from "@mui/icons-material/CloseRounded";
+import BookmarkBorderIcon from "@mui/icons-material/BookmarkBorder";
+import {
+  isMobile,
+  runningInFirefoxDesktop,
+  runningInChromeDesktop,
+} from "../utils/misc/browserDetection";
+
+export default function RedirectionNotificationForUnsupportedBrowsers({
+  api,
+  article,
+  setIsArticleSaved,
+  toggleRedirectionCheckboxSelection,
+  selectedDoNotShowRedirectionModal_Checkbox,
+  handleCloseAndSaveVisibilityPreferences,
```

Comment:
the comment suggest a different name for this method? i.e. something along the lines of `openArticleAndSavePreferences`? Then again, it's not nice to have a method that does two different things like that. I wonder if it would be just nicer to leave the method to do simply `openArticle` and always save the preferences when one checks/unchecks the checkbox? Alternatively, keep the behavior but just name it `openArticle`. And add a note there saying that we also save the preference when we open. 

<img width="879" alt="image" src="https://github.com/zeeguu/web/assets/464519/a3b21d6c-98b1-45d0-90d0-ffd6542200a5">

Reply:
>I wonder if it would be just nicer to leave the method to do simply openArticle and always save the preferences when one checks/unchecks the checkbox? Alternatively, keep the behavior but just name it openArticle.

This one is tricky. Unfortunately, we can’t save the visibility preferences immediately when the user checks/unchecks the "do not show" checkbox - I tried this solution in the past. It resulted in the modal disappearing immediately when the "do not show" checkbox is checked (as it affected conditional rendering in the `titleLink(article)` function in the ArticlePreview). This is why I had to **separate** it so that **the checkbox takes care of toggling visibility preferences only**, but **they are saved to the local storage when the user clicks the modal button (Go to article / Save article).** 

-----
I fully agree that `handleCloseAndSaveVisibilityPreferences` it is a very long name for the function, but I am not sure it should be renamed to `openArticle`. The comment in the code from the screenshot refers to the behavior of the whole button and the link to the `{article.url}`, not only to the function. I should have placed it above the `<a></a>` tag.

<br>

**The itself function takes care of**:
- setting the modal visibility preference state that updates the local storage settings thanks to the `handleModalVisibilityPreferences()` inside
- closing the modal

```
  function handleCloseAndSaveVisibilityPreferences() {
    handleModalVisibilityPreferences();
    handleCloseRedirectionModal();
  }
```

**Use cases of this function:** 

- **For supported browsers**:
The `handleCloseAndSaveVisibilityPreferences` function is attached to the button opening articles so that when the user goes to the article, their modal visibility settings are saved, and the modal is no longer open when they return to Zeeguu

- **For unsupported browsers**:
The `handleCloseAndSaveVisibilityPreferences` function runs together with: the handleSaveArticle in the Save Article button so that modal visibility preferences are saved, the article is added to saves, and the modal is closed